src/components/SegmentedControl/index.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { ReactNode } from 'react';
+import PropTypes from 'prop-types';
+import RenderIf from '../RenderIf';
+import { BaseProps } from '../types';
+import {
+    StyledSegmentedControlContainer,
+    StyledSegmentedControlLabel,
+    StyledSegmentedControlOptionsWrapper,
+} from './styled';
+import SegmentedControlOption from './SegmentedControlOption';
+
+interface Option {
+    label: ReactNode;
+    value: string;
+    icon?: ReactNode;
+}
+
+interface Props extends BaseProps {
+    label?: string;
+    variant?: 'default' | 'brand';
+    value?: string;
+    options?: Option[];
+    onChange?: (event: any) => void;
+    isDisabled?: boolean;
+    isFullWidth?: boolean;
+}
+
+/**
+ * SegmentedControl component controls user interacion with a particular process.
+ */
+
+const SegmentedControl: React.FC<Props> = props => {
+    const {
+        style,
+        label,
+        variant = 'default',
+        value,
+        options = [],
+        onChange = () => {},
+        isDisabled,
+        isFullWidth,
+    } = props;
+    const [active, setActive] = React.useState(value);
+
+    let handleOnChange = (value: string) => {
+        setActive(value);
+        onChange(value);
+    };
+
+    let hasLabel = !!label;
+    return (
+        <StyledSegmentedControlContainer
+            isFullWidth={isFullWidth}
+            style={style}
+        >
+            <RenderIf isTrue={hasLabel}>
+                <StyledSegmentedControlLabel>
+                    {label}
+                </StyledSegmentedControlLabel>
+            </RenderIf>
+            <StyledSegmentedControlOptionsWrapper isDisabled={isDisabled}>
+                {options.map((option, index) => (
+                    <SegmentedControlOption
+                        key={index}
```

Comment:
react normally warns when using the index as key directly, instead construct one like:
```const key = `option-${index}````